compiler/rustc_mir_transform/src/const_prop.rs

```diff
@@ -308,14 +308,14 @@ impl<'mir, 'tcx> interpret::Machine<'mir, 'tcx> for ConstPropMachine<'mir, 'tcx>
     }
 
     #[inline(always)]
-    fn stack(
+    fn stack<'a>(
         ecx: &'a InterpCx<'mir, 'tcx, Self>,
     ) -> &'a [Frame<'mir, 'tcx, Self::PointerTag, Self::FrameExtra>] {
```

Comment:
Looks like here and in some cases below the lifetime (`'a`) can be omitted?
(Because single input lifetime relates with an output lifetimes automatically.)

Reply:
I went and tried it, but there are enough lifetimes going on here that is needs the name:
> help: this function's return type contains a borrowed value, but the signature does not say which one of `ecx`'s 3 lifetimes it is borrowed from